### PR TITLE
feat: add basic commands

### PR DESCRIPTION
**Pull request by hboissel**
both commands added to a module commands. Could be relevant to move print kernel stack there as well.
The reboot command jump to the label created in the initial assembly.
The halt command runs hlt.
Shutdown send a shutdown to qemu so works only on qemu since shutdown is a power management action, it is specific to the hardware.

### CODE REVIEW

**Comment by alisterd51** [comment]
This addition is a feature: the merge commit and the name of the MR should start with `feat:` instead of `build:`.

**Comment by hboissel** [comment]
Screenshot from 2024-07-24 13-45-56

**Review by alisterd51 — CHANGES_REQUESTED** [review]

File: src/commands/print_gdt.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+use core::arch::asm;
+
+struct SegmentDescriptor {
+    base: u32,
+    limit: u32,
+    flags: u8,
+    access_byte: u8,
+}
+
+fn parse_segment_descriptor(descriptor: u64) -> SegmentDescriptor {
```
[the remark this answers is not in the document]
Comment (hboissel) [reply src/commands/print_gdt.rs:11]: image

**Comment by alisterd51** [comment]
can you update the readme regarding the whole GDT part?

File: README.md [hunk 2]
```diff
@@ -64,24 +64,24 @@ qemu-system-x86_64 -cdrom gratos.x86.iso
...
   - [ ] Bonus:
     - [ ] Assuming your keyboard work correctly in your Kernel, and you able to catch an entry, let’s code a Shell !
         Not a POSIX Shell, just a minimalistic shell with a few commands, for debugging purposes.
     - [ ] For example, you could implement the print-kernel-stack-thing in this shell, and some other things like reboot, halt and such.
-      - [ ] print-kernel-stack-thing
-      - [ ] reboot
-      - [ ] halt
+      - [x] print-kernel-stack-thing
+      - [x] reboot
+      - [x] halt
```
Comment (alisterd51) [inline README.md:85]: These features are not yet implemented. They will have to be added after the basic shell merge.

File: src/commands/print_gdt.rs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    println!(
+        "{}Limit: {:#010X}{}",
+        driver::tty::FG_BRIGHT_CYAN,
+        limit,
+        driver::tty::RESET
+    );
+}
+
+pub fn print_gdt() {
+    let mut gdt: [u16; 2] = [0; 2];
```
Comment (alisterd51) [inline src/commands/print_gdt.rs:143]:
```rust
    let mut gdt = [0u16; 2];
```

**Review by alisterd51 — CHANGES_REQUESTED** [review]

**Pull request merged**